chef/cookbooks/barclamp/libraries/barclamp_library.rb

```diff
@@ -312,7 +317,24 @@ def <=>(other)
           self.name <=> other.name
         end
 
+        # is the current disk already claimed? then use the claimed unique_name
+        def unique_name_already_claimed_by
+          cm = @node[:crowbar_wall][:claimed_disks].find do |claimed_name, v|
+            self.link_to_name?(claimed_name)
+          end || []
+          cm.first
+        end
+
         def unique_name
+          # check first if we have already a claimed disk which points to the same
+          # device node. if so, use that as "unique name"
+          already_claimed_name = self.unique_name_already_claimed_by
+          unless already_claimed_name.nil?
+            Chef::Log.info("Use #{already_claimed_name} as unique_name " \
```

Comment:
Done in https://github.com/toabctl/barclamp-deployer/commit/ab869cbde4dd2f74a98d58a1c86b84c4d2399654